### PR TITLE
[Documentation] Describe Sylius plugins

### LINKED ISSUE #7851: Docs about 'SyliusPluginTrait'
The current docs do not describe anything about using the `SyliusPluginTrait` yet, also the Sylius skeleton bundle doesn't use it. I would write them myself, but I'm not 100% sure yet how it is intended to be used. Besides that, I think there was also some discussion whether it should be Plugin or Bundle suffix, which is probably decided upon using this trait.

Related commit: https://github.com/Sylius/Sylius/commit/0010f551b831f3df5ec57a561b4c6ba8916f2af0

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | fixes #7851 |
| License         | MIT |

### CODE REVIEW

File: docs/extensions/creating-extension.rst [hunk 1]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
 .. _`Behat`: http://behat.org/en/latest/
 .. _`phpspec`: http://www.phpspec.net/en/stable/
 .. _`PHPUnit`: https://phpunit.de/
+
+5. Naming convention
+--------------------
+
+Beside the way how you would like to create a plugin(based on our skeleton or by your own) there are a few naming conventions that should be followed:
```
Comment (Zales0123) [inline docs/extensions/creating-extension.rst]: Missing comma before `(`.

File: docs/extensions/creating-extension.rst [hunk 2]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
 .. _`Behat`: http://behat.org/en/latest/
 .. _`phpspec`: http://www.phpspec.net/en/stable/
 .. _`PHPUnit`: https://phpunit.de/
+
+5. Naming convention
```
Comment (Zales0123) [inline docs/extensions/creating-extension.rst]: convention**s**?

File: docs/extensions/creating-extension.rst [hunk 3]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+--------------------
+
+Beside the way how you would like to create a plugin(based on our skeleton or by your own) there are a few naming conventions that should be followed:
+
+ * Bundle class must have a `Plugin` suffix instead of `Bundle` in its name (e.g. InvoicePlugin).
+ * Bundle class must use the `Sylius\Bundle\CoreBundle\Application\SyliusPluginTrait`.
+ * The name of extension in DependencyInjection folder follow the regular Symfony rules (e.g. InvoiceExtension).
```
Comment (Zales0123) [inline docs/extensions/creating-extension.rst]: missing **must** before `follow`.

File: docs/extensions/creating-extension.rst [hunk 4]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+ * The plugin shouldn't have Sylius prefix in the plugin name. `Plugin` suffix in terms of bundles is unique for Sylius at the moment.
```
Comment (Zales0123) [inline docs/extensions/creating-extension.rst]: `plugin name` -> `its name`?

File: docs/extensions/creating-extension.rst [hunk 5]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+
+The following rules are applied to all bundles which will provide an integration with whole Sylius platform(whole Sylius/SyliusCoreBundle in vendors).
```
Comment (Zales0123) [inline docs/extensions/creating-extension.rst]: `whole` -> `the whole`

Comment (Zales0123) [inline docs/extensions/creating-extension.rst]: missing comma before `(`

File: docs/extensions/creating-extension.rst [hunk 6]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+Reusable components for whole Symfony community which will be based just on some Sylius bundles should be follow regular Symfony conventions.
```
Comment (Zales0123) [inline docs/extensions/creating-extension.rst]: `should be follow` -> `should follow`

File: docs/extensions/creating-extension.rst [hunk 7]
```diff
@@ -1,17 +1,17 @@
-How to create an extension for Sylius?
-======================================
+How to create a plugin for Sylius?
+==================================
 
-Sylius extension is nothing more but a regular Symfony bundle adding custom behaviour to the default Sylius application.
+Sylius plugins is nothing more but a regular Symfony bundle adding custom behaviour to the default Sylius application.
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: *`plugin` - singular

File: docs/extensions/creating-extension.rst [hunk 8]
```diff
@@ -1,17 +1,17 @@
...
 
-The best way to create your own extension is to use `Sylius bundle skeleton <https://github.com/Sylius/BundleSkeleton>`_,
+The best way to create your own extension is to use `Sylius plugin skeleton <https://github.com/Sylius/PluginSkeleton>`_,
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: your own **plugin**

File: docs/extensions/creating-extension.rst [hunk 9]
```diff
@@ -32,7 +32,7 @@ and the testing infrastructure in ``tests``.
 ---------------------------------------------------
 
 In most cases you don't want your Sylius extension to greet customer like it is now, so feel free to remove unnecessary
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: <del> extension </del> -> **plugin** in the whole document, please.

File: docs/extensions/creating-extension.rst [hunk 10]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
 .. _`Behat`: http://behat.org/en/latest/
 .. _`phpspec`: http://www.phpspec.net/en/stable/
 .. _`PHPUnit`: https://phpunit.de/
+
+5. Naming conventions
+---------------------
+
+Beside the way how you would like to create a plugin, (based on our skeleton or by your own) there are a few naming conventions that should be followed:
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: `Besides the way you are creating plugins (based on our skeleton or on your own), ...`

File: docs/extensions/creating-extension.rst [hunk 11]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+
+ * Bundle class must have a `Plugin` suffix instead of `Bundle` in its name (e.g. InvoicePlugin).
+ * Bundle class must use the `Sylius\Bundle\CoreBundle\Application\SyliusPluginTrait`.
+ * The name of the extension in DependencyInjection folder must follow the regular Symfony rules (e.g. InvoiceExtension).
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: what does it mean? Shouldn't it be `plugin`?

File: docs/extensions/creating-extension.rst [hunk 12]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+ * The plugin shouldn't have Sylius prefix in its name. `Plugin` suffix in terms of bundles is unique for Sylius at the moment.
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: I know, I know. But I am really sad, it's been decided so.

File: docs/extensions/creating-extension.rst [hunk 13]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+
+The following rules are applied to all bundles which will provide an integration with the whole Sylius platform (whole Sylius/SyliusCoreBundle in vendors).
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: shouldn't it be `sylius/sylius`?

File: docs/extensions/creating-extension.rst [hunk 14]
```diff
@@ -53,3 +53,16 @@ to ensure your extension's extraordinary quality.
...
+Reusable components for whole Symfony community which will be based just on some Sylius bundles should follow regular Symfony conventions.
```
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: the whole Symfony community, which will be based

**Comment by TheMadeleine** [comment]
The name of the file should be changed also. and the directory. And the `.index.rst` file needs to be updated. And the `list.rst` also. ⚠️

File: docs/extensions/creating-extension.rst (hunk 11 above)
Comment (lchrusciel) [inline docs/extensions/creating-extension.rst]: I don't know how to say this more exactly. `Extension` file from DI folder(which has to have `Extension` prefix) shouldn't have `Plugin` inside its name. See this example:
Plugin name is AcmeExample**Plugin**
Plugin
Extension name is AcmeExample**Extension**
Extension

File: docs/extensions/creating-extension.rst (hunk 12 above)
Comment (lchrusciel) [inline docs/extensions/creating-extension.rst]: 😭

File: docs/extensions/creating-extension.rst (hunk 11 above)
Comment (TheMadeleine) [inline docs/extensions/creating-extension.rst]: Okay, I get it, but... seems really confusing. : ]

File: docs/plugins/creating-plugin.rst [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+.. code-block:: bash
+
+    $ composer create-project sylius/plugin-skeleton MyPlugin
+
+.. note::
+
+    The plugin can be created anywhere, not only inside Sylius application, because it already has the test environment inside.
+
+2. Get familiar with basic plugin design.
+--------------------------------------------
```
Comment (TheMadeleine) [inline docs/plugins/creating-plugin.rst]: the line below the title seems too long ;>

File: docs/plugins/index.rst [hunk 16]
```diff
@@ -0,0 +1,35 @@
+Sylius Plugins
+==============
+
+Sylius as a platform has a lot of space for various customizations and extensions. It aims to provide a simple schema for
+developing extensions (plugins). Anything you can imagine can be implemented and added to the Sylius framework as a plugin.
```
Comment (TheMadeleine) [inline docs/plugins/index.rst]: `developing plugins.`

File: docs/plugins/index.rst [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+
+.. toctree::
+    :hidden:
+
+    creating-extension
```
Comment (TheMadeleine) [inline docs/plugins/index.rst]: creating-plugin

File: docs/plugins/index.rst [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+are built on top of the Sylius Core.
+
+Exemplary features may be: Social media buttons, newsletter, wishlists etc.
+
+Plugin Development Flow
+-----------------------
+
+Since Sylius is an open-source platform, there is a certain flow in order for the plugin to become officially adopted by the community.
+
+**1.** Develop the plugin using :doc:`the official Plugin Development guide </extensions/creating-plugin>`.
```
Comment (TheMadeleine) [inline docs/plugins/index.rst]: `</plugins/creating-plugin>`

**Comment by pjedrzejewski** [comment]
Thanks Łukasz!

**Pull request merged**